tests/unit/src/test/scala/tests/debug/DotEnvFileParserSuite.scala

```diff
@@ -0,0 +1,155 @@
+    assertDiffEqual(parse("KEY="), Map("KEY" -> ""))
+  }
+
+  test("parse values separated by :") {
+    assertDiffEqual(parse("KEY:value"), keyValue)
+  }
+
+  test("parse values separated by : with spaces around separator") {
+    assertDiffEqual(parse("KEY:  value"), keyValue)
+    assertDiffEqual(parse("KEY  :value"), keyValue)
+    assertDiffEqual(parse("KEY  :  value"), keyValue)
+  }
+
+  test("parse unquoted values") {
+    assertDiffEqual(parse("KEY=value"), keyValue)
+  }
+
+  test("strip unquoted values") {
+    assertDiffEqual(parse("KEY=value  "), keyValue)
+  }
+
+  test("parse unquoted values with spaces around separator") {
+    assertDiffEqual(parse("KEY  =value"), keyValue)
+    assertDiffEqual(parse("KEY=  value"), keyValue)
+    assertDiffEqual(parse("KEY  =  value"), keyValue)
+  }
+
+  test("parse unquoted values with leading and trailing spaces") {
+    assertDiffEqual(parse("  KEY=value"), keyValue)
+    assertDiffEqual(parse("KEY=value  "), keyValue)
+    assertDiffEqual(parse("  KEY=value  "), keyValue)
+  }
+
+  test("parse single quoted values") {
+    assertDiffEqual(parse("KEY='value'"), keyValue)
+  }
+
+  test("strip single quoted values") {
+    assertDiffEqual(parse("KEY='  value'"), keyValue)
+    assertDiffEqual(parse("KEY='value  '"), keyValue)
+    assertDiffEqual(parse("KEY=' value  '"), keyValue)
+  }
+
+  test("parse double quoted values") {
+    assertDiffEqual(parse("""KEY="value""""), keyValue)
+  }
+
+  test("strip double quoted values") {
+    assertDiffEqual(parse("""KEY="  value""""), keyValue)
+    assertDiffEqual(parse("""KEY="value  """"), keyValue)
+    assertDiffEqual(parse("""KEY="  value  """"), keyValue)
+  }
+
+  test("parse escaped double quotes") {
+    assertDiffEqual(parse("""KEY="\"quoted\"""""), Map("KEY" -> "\"quoted\""))
+  }
+
+  test("parse export keyword") {
+    assertDiffEqual(parse("export KEY=value"), keyValue)
+  }
+
+  test("ignore lines with no assignment") {
+    assertDiffEqual(parse("KEY"), Map.empty)
+  }
+
+  test("ignore lines with comments") {
+    assertDiffEqual(parse("#KEY=value"), Map.empty)
+  }
+
+  test("ignore empty lines") {
+    assertDiffEqual(parse(" \nKEY=value"), keyValue)
+  }
+
+  test("allow # and spaces in unquoted values") {
+    assertDiffEqual(
+      parse("KEY=v a l u e # not a comment"),
+      Map("KEY" -> "v a l u e # not a comment")
+    )
+  }
+
+  test("ignore inline comments after single quoted values") {
+    assertDiffEqual(parse("KEY='value' # comment"), keyValue)
+  }
+
+  test("ignore inline comments after double quoted values") {
+    assertDiffEqual(parse("""KEY="value" # comment"""), keyValue)
+  }
+
+  test("allow # in quoted values") {
+    assertDiffEqual(parse("""export KEY="v#l#e""""), Map("KEY" -> "v#l#e"))
+  }
+
+  test("parse multi-line single quoted values") {
+    val content = """KEY='line1
+                    |line2'""".stripMargin
+    assertDiffEqual(parse(content), Map("KEY" -> "line1\nline2"))
```

Comment:
It seems the tests with `\n` are failing on windows. It might have something to do with additional `\r` there.

We could try doing `val separator = if (isWindows) "\r\n" else "\n"`, but not 100% if that will help. I try checking it out on windows if needed.

Reply:
I completely forgot about Windows - I think it's a case of substituting all `\r\n` with `\n`. I'll try to fix that later today.

Reply:
Thanks!